lib/Sync/SharedMemoryParcel.php

```diff
@@ -312,10 +312,11 @@ private function __clone()
     }
 
     /**
-     * Private method to prevent serialization.
+     * Prevent serialization.
      */
-    private function __sleep()
+    public function __sleep()
     {
+        throw new SharedMemoryException('A shared memory parcel cannot be serialized!');
     }
```

Comment:
Interesting, I thought `private function __sleep()` would prevent serialization, but it actually only issues a warning on older versions, too. https://3v4l.org/F4Eu8

I'm not sure which kind of exception we should throw here. I think I prefer to just throw an `\Error`.

Reply:
> I think I prefer to just throw an \Error.

Same.